dashboard/test/models/concerns/user/level_progressable_test.rb

```diff
@@ -0,0 +1,671 @@
+require 'test_helper'
+
+class LevelProgressableTest < ActiveSupport::TestCase
+  include Minitest::RSpecMocks
+
+  def create_level_group(sub_level_name)
```

Comment:
I don't love that this code is now being duplicated from `user_test.rb`. Would it be feasible to use [our existing LevelGroup factory](https://github.com/code-dot-org/code-dot-org/blob/d317465fac6cd0d16c84b8e2fb87e11f5e1990b1/dashboard/test/factories/factories.rb#L1488) here instead? It might involve adding a trait that allows us to specify the name of the sublevel, but that sounds like it'd have general value.

Reply:
I agree with you on this. As this was the first concern I worked with, I focused first on moving the tests. I'll try to add a factory, or remove the duplicated code at once.

Reply:
It would also be totally reasonable to do that as a follow-up